### ISSUE TITLE
[NBug] Object reference not set to an instance of an object.

### ISSUE THREAD

**Issue by TymurGubayev** [gitextensions/gitextensions#8998]
:warning: The sections below must be filled in and this text must be removed or the issue will be closed.

## Error Details
```
System.NullReferenceException: Object reference not set to an instance of an object.
   at ICSharpCode.TextEditor.Document.LineSegmentTree.GetNode(Int32 index)
   at ICSharpCode.TextEditor.Document.DefaultDocument.GetLineSegment(Int32 line)
   at GitUI.Editor.Diff.DiffHighlightService.ProcessLineSegment(IDocument document, Int32& line, LineSegment lineSegment, String prefixStr, Color color)
   at GitUI.Editor.Diff.DiffHighlightService.AddPatchHighlighting(IDocument document)
   at GitUI.Editor.FileViewerInternal.SelectionManagerSelectionChanged(Object sender, EventArgs e)
   at System.EventHandler.Invoke(Object sender, EventArgs e)
   at ICSharpCode.TextEditor.Document.SelectionManager.OnSelectionChanged(EventArgs e)
   at ICSharpCode.TextEditor.Document.SelectionManager.ClearWithoutUpdate()
   at ICSharpCode.TextEditor.Document.SelectionManager.ClearSelection()
   at ICSharpCode.TextEditor.Document.SelectionManager.RemoveSelectedText()
   at ICSharpCode.TextEditor.Actions.Delete.DeleteSelection(TextArea textArea)
   at ICSharpCode.TextEditor.Actions.Delete.Execute(TextArea textArea)
   at ICSharpCode.TextEditor.TextArea.ExecuteDialogKey(Keys keyData)
   at ICSharpCode.TextEditor.TextArea.ProcessDialogKey(Keys keyData)
   at System.Windows.Forms.Control.PreProcessMessage(Message& msg)
   at System.Windows.Forms.Control.PreProcessControlMessageInternal(Control target, Message& msg)
   at System.Windows.Forms.Application.ThreadContext.PreTranslateMessage(MSG& msg)
```

## Additional information
Trying to edit a .gitignore to exclude a file with colon in it's name - which is valid on Linux but not on Windows

## Environment
- Git Extensions 3.4.3.9999
- Build d4b0f48bbf3e39a71f5d7ff5231be5678d4aa0f1
- Git 2.30.1.windows.1
- Microsoft Windows NT 10.0.19042.0
- .NET Framework 4.8.4341.0
- DPI 96dpi (no scaling)

**Comment by RussKie** [comment]
I think this was fixed. Can you try the latest dev version?

**Comment by TymurGubayev** [comment]
I've got portable version 3.5.0.11443 from https://ci.appveyor.com/project/gitextensions/gitextensions/builds/38227685/artifacts and 3.6.0.11442 from https://ci.appveyor.com/project/gitextensions/gitextensions/builds/38223415/artifacts, both have the same problem.

I've also found a reliable way to reproduce:

1. Open any repository
2. Select Repository -> Edit .gitignore
3. Type in "a\b:c"
4. Select "a"
5. Press "*" (expecting to get "*\b:c")
6. Get an error.

If there is already such a file in the repository, then "Open this revision (temp file)" shows an error, but "Open this revision with... (temp file)" crashes with `System.NotSupportedException`.
